libsolidity/formal/CHC.cpp

```diff
@@ -687,10 +701,19 @@ pair<smtutil::Expression, smtutil::Expression> CHC::arithmeticOperation(
 	}
 	else if (intType->isSigned())
 	{
+		addVerificationTarget(
+			&_expression,
+			VerificationTarget::Type::Underflow,
+			errorFlag().currentValue()
+		);
+		target = values.second < intType->minValue() && errorFlag().currentValue() == errorId;
+		m_context.addAssertion((errorFlag().currentValue() == previousError) || *target);
+
+		previousError = errorFlag().currentValue();
+		errorFlag().increaseIndex();
```

Comment:
Maybe we can extract this functionality to a helper method, which would ensure that the errorFlag is used and set correctly when new verification target is added.

Reply:
You mean a function that increase the index but returns the previous one?

Reply:
I have now created a helper function, createAndAddVerificationTarget, in the CHC::arithmeticOperation. I was not sure if it should be a member function of CHC, for now I have created it only as a lambda and use it only in that one method. It wraps up the necessary handling of the error flag.

Reply:
I could actually use it in one more place, at the beginning of that method.

Reply:
And it could be used also in another place, I am going to turn it in a member function.